dotCMS/src/main/java/com/dotcms/saml/DotSamlConfigurationServiceImpl.java

```diff
@@ -0,0 +1,230 @@
+    private static final String NOT_FOUND_ERROR = "Property Name not Found: ";
+    /**
+     * To set into the init map context, the absolute path of the default properties for SAML.
+     */
+    public final static String DOT_SAML_DEFAULT_PROPERTIES_CONTEXT_MAP_KEY = "dotSamlDefaultPropertiesContextMapKey";
+
+    private AtomicBoolean init = new AtomicBoolean(false);
+    private final Map<String, String> defaultProperties = new ConcurrentHashMap<>(this.createInitialMap());
+
+    @Override
+    public void initService(final Map<String, Object> contextMap) {
+
+        if (!this.init.get()) {
+
+            this.internalInit(contextMap);
+        }
+    }
+
+    private synchronized void internalInit(final Map<String, Object> contextMap) {
+
+        final String dotSamlDefaultPropertiesValue = (String)contextMap.get(DOT_SAML_DEFAULT_PROPERTIES_CONTEXT_MAP_KEY);
+
+        if (null == dotSamlDefaultPropertiesValue) {
+
+            Logger.warn(this, DOT_SAML_DEFAULT_PROPERTIES_CONTEXT_MAP_KEY + " must be set on the argument context map");
+        } else {
+
+            final File dotSamlDefaultPropertiesFile = new File(dotSamlDefaultPropertiesValue);
+
+            if (!dotSamlDefaultPropertiesFile.exists() || !dotSamlDefaultPropertiesFile.canRead()) {
+
+                Logger.warn(this, "The " + dotSamlDefaultPropertiesValue + " does not exists or can not read");
+            } else {
+
+                final Properties properties = new Properties();
+
+                try (InputStream input = Files.newInputStream(dotSamlDefaultPropertiesFile.toPath())) {
+
+                    properties.load(input);
+                } catch (IOException ex) {
+                    // Since this is optional, it is valid to not have the file.
+                    // Log and go on.
+                    Logger.warn(this, UNABLE_TO_READ_FILE + dotSamlDefaultPropertiesValue);
+
+                }
+
+                properties.forEach((key, value) -> {
+
+                    final SamlName samlName = SamlName.findProperty((String)key);
+                    if (null != samlName) {
+
+                        this.defaultProperties.put(samlName.getPropertyName(), (String)value);
+                    }
+                });
+
+                this.init.set(true);
+            }
+        }
+    }
+
+
+    @Override
+    public String getConfigAsString(final IdentityProviderConfiguration identityProviderConfiguration, final SamlName samlName) {
+
+        try {
+
+            final String value = identityProviderConfiguration.containsOptionalProperty(samlName.getPropertyName())?
+                    (String) identityProviderConfiguration.getOptionalProperty(samlName.getPropertyName()):
+                    this.getDefaultStringParameter(samlName);
+
+            Logger.debug(this,
+                    ()-> "Found " + samlName.getPropertyName() + " : " + ((value == null) ? "null" : value));
```

Comment:
Codacy found an issue: [The String literal " : " appears 4 times in this file; the first occurrence is on line 99](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)